jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/listeners/Asn1DecodedDataListener.java

```diff
@@ -0,0 +1,82 @@
+package us.dot.its.jpo.ode.kafka.listeners;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.annotation.KafkaHandler;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.messaging.handler.annotation.Headers;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Component;
+import us.dot.its.jpo.ode.coder.OdeMapDataCreatorHelper;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.kafka.OdeKafkaProperties;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeLogMetadata;
+import us.dot.its.jpo.ode.util.XmlUtils;
+
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+@Component
+@KafkaListener(id = "Asn1DecodedDataListener", topics = "${ode.kafka.topics.asn1.decoder-output}", containerFactory = "tempFilteringKafkaListenerContainerFactory")
```

Comment:
1. That's still up in the air. The simplest implementation from a migration perspective would definitely be to support all message types received on the decoder output topic. There are other more targeted options, however, that I think may be smarter. There is a way in the Spring Kafka library/framework to direct specific message types to specific listeners using the same topic. That would allow us to keep the consumption logic separate for each message type, but it would be more work and have more moving parts. So, like I said, it's still up in the air
2. This listener uses the [tempFilteringKafkaListenerContainerFactory](https://github.com/CDOT-CV/jpo-ode/blob/0fd3d4caf1f65e475f5f7fdd7381b7dbd97205f6/jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/KafkaConsumerConfig.java#L74) to filter out all messages not of MAP type. It only consumes MAP messages while the Asn1DecodedDataRouter and related classes still consume the other non-MAP data types. As this effort evolves, the filter will go away, but it's how we're supporting both the hand-rolled and spring kafka consumers
3. That's a really good point and good question. I haven't decided yet, but it sounds like the best course of action is to just use this new listener for all data types on the decoded data output topic. If we find that we want to use multiple listeners, then yeah, we'd use different groupIds for each data type related consumer.

Reply:
Are you planning on tackling the containerFactory implementation for additional types as they come on then?

Reply:
@payneBrandon, that was the tentative plan when I implemented this. After thinking more about this, and thanks to this comment chain, I think the better course of action would be to migrate all of the consumption managed in [Asn1DecodedDataRouter](https://github.com/CDOT-CV/jpo-ode/blob/f309463e147878cf1252c13862676bcdbe8a0644/jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1DecodedDataRouter.java#L36) to [Asn1DecodedDataListener](https://github.com/CDOT-CV/jpo-ode/blob/369cc4bd627ab047c55d251cd82043c89485f333/jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/listeners/Asn1DecodedDataListener.java#L24) in one work item. That way, we can get rid of the filter in one unit of work,  get rid of the [tempFilteringKafkaListenerContainerFactory](https://github.com/CDOT-CV/jpo-ode/blob/0fd3d4caf1f65e475f5f7fdd7381b7dbd97205f6/jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/KafkaConsumerConfig.java#L74) altogether, and everything will be unified under one consumer again

Reply:
Okay this all makes sense. I like the plan of action for incorporating the other data feeds.